gnd/src/main/java/com/google/android/gnd/ui/offlinebasemap/OfflineBaseMapsViewModel.java

```diff
@@ -21,41 +21,64 @@
 import androidx.lifecycle.LiveDataReactiveStreams;
 import com.google.android.gnd.model.basemap.OfflineBaseMap;
 import com.google.android.gnd.repository.OfflineBaseMapRepository;
+import com.google.android.gnd.rx.annotations.Cold;
 import com.google.android.gnd.ui.common.AbstractViewModel;
 import com.google.android.gnd.ui.common.Navigator;
 import com.google.common.collect.ImmutableList;
+import io.reactivex.Flowable;
 import javax.inject.Inject;
+import timber.log.Timber;
 
 /**
  * View model for the offline area manager fragment. Handles the current list of downloaded areas.
  */
 public class OfflineBaseMapsViewModel extends AbstractViewModel {
 
-  private LiveData<ImmutableList<OfflineBaseMap>> offlineAreas;
-  private LiveData<Integer> noAreasMessageVisibility;
+  @Cold(terminates = false, errors = false)
+  private final LiveData<ImmutableList<OfflineBaseMap>> offlineAreas;
+
+  @Cold(terminates = false, errors = false)
+  private final LiveData<Integer> noAreasMessageVisibility;
+
   private final Navigator navigator;
 
   @Inject
   OfflineBaseMapsViewModel(Navigator navigator, OfflineBaseMapRepository offlineBaseMapRepository) {
     this.navigator = navigator;
-    this.offlineAreas =
-        LiveDataReactiveStreams.fromPublisher(
-            offlineBaseMapRepository.getOfflineAreasOnceAndStream());
+    Flowable<ImmutableList<OfflineBaseMap>> offlineAreas =
```

Comment:
Throwing this out there for future work: What if we roll the `doOnError`, `onErrorReturnItem`, and `fromPublisher` into a `Converter` returned by a function, something like `handleErrors`. Then we could do use it to log and should a message in the UI in a consistent way, e.g.
```java
       offlineBaseMapRepository
            offlineBaseMapRepository.getOfflineAreasOnceAndStream())	            
              .getOfflineAreasOnceAndStream()
              .compose(handleErrors("Error loading base map", ImmutableList.of()));
```

`handleErrors` could both log the exception to`Timber.e` with the provided message, and show a toast. If that's helpful I can create a bug and do this in a followup.

Reply:
Sounds useful to me! I created a bug: https://github.com/google/ground-android/issues/743